packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-options.page.tsx

```diff
@@ -60,15 +60,25 @@ export const getServerSideProps: GetServerSideProps = async ({
     req,
     res,
     async (body: any) => {
-      options = body.options;
+      options = Object.keys(body).reduce((array, key) => {
+        if (key.startsWith('options')) {
+          array.push(...body[key]);
+        }
+
+        if (key.startsWith('delete_')) {
+          const deleteOptionIndex = Number(key.split('_')[1]);
+          array.splice(deleteOptionIndex, 1);
+        }
+        return array;
+      }, [] as string[]);
 
       if ('add-another-option' in body) {
         options.push('');
 
         return {
           data: 'OPTION_ADDED',
         };
-      } else {
+      } else if ('save-question' in body) {
```

Comment:
Given there are a couple more scenarios here we could use a switch statement to improve readability? what do you think?

Reply:
the switch doesn't help in this case as we're checking for keys in an object - it means we need to use `switch(true)` like in the other place we're doing this which i intend to refactor out anyway

Reply:
From what i've read using `switch(true)` is pretty widely accepted and not considered bad practice. Would say the readability of this is pretty subjective too as neither is right or wrong. 

If it was any more than 3 cases i'd say go for switch over if else but it's up to you @paul-lawlor-tco

Reply:
appreciate that the switch(true) is a bit strange, although having the labels explicitly written out for each case makes it easier to read what's going on imo (regardless of the extra indentation)

will keep it as the switch for now and if you're refactoring then feel free! 

Thank you both for the input on this :)